### ISSUE TITLE
Remote schema JSON key escaping issues

### ISSUE THREAD

**Issue by namoscato** [hasura/graphql-engine#8200]
### Version Information

Server Version: v2.2.0

### Environment

OSS and EE (`hasuraci/graphql-engine-pro:v2.2.0-pro.1`)

### What is the expected behaviour?

JSON field keys included in the response of a remote schema should be properly escaped

### Keywords

remote schema, json, property, key, escaping, escape

### What is the current behaviour?

* When JSON object keys include problematic characters such as backslash (`\`) or double quote (`"`), Hasura returns a 200 status code with an invalid JSON response
* This does not seem to affect JSON object _values_ – just _keys_
* Possibly related to https://github.com/hasura/graphql-engine/issues/7543

### How to reproduce the issue?

1. Given a GraphQL server resolver implementation, i.e. we're using [Apollo Server](https://www.apollographql.com/docs/apollo-server/) with its schema generated by [GraphQL Nexus](https://nexusjs.org/):

    ```ts
    export const JsonTestQuery = queryField("json_test", {
      type: objectType({
        name: "json_test_response",
        definition(t) {
          t.json("json_field");
        },
      }),
      async resolve() {
        return {
          json_field: {
            "foo\\": "bar",
            '"foo"': "bar",
          },
        };
      },
    });
    ```

1. Hit up the GraphQL server directly, and see that the response is valid JSON:

    ```graphql
    query JsonTest {
      json_test {
        json_field
      }
    }
    ```
    
    ```json
    {"data":{"json_test":{"json_field":{"foo\\":"bar","\"foo\"":"bar"}}}}
    ```

1. Hit up the same query through Hasura's remote schema configuration, and see that the response is invalid JSON:

    ```
    {"data":{"json_test":{"json_field":{"foo\":"bar",""foo"":"bar"}}}}
    ```

    Specifically, neither the backslash (`\`) nor double quote (`"`) characters are properly escaped. Unsure if this is an issue for other problematic characters.

### Any possible solutions?

It's unclear why there is really _any_ transformation applied to JSON field values returned from a remote schema

**Comment by robx** [comment]
This appears to be a bug in `Hasura.Encode`:

```
> putStrLn $ Data.Aeson.encode $  Data.Map.fromList [("a\\b\"c", A.String "foo")]
{"a\\b\"c":"foo"}
> putStrLn $ encJToLBS $ encJFromOrderedValue $ Data.Aeson.Ordered.Object $ Data.Aeson.Ordered.fromList [("a\\b\"c", Data.Aeson.Ordered.String "foo")]
{"a\b"c":"foo"}
```

**Comment by abooij** [comment]
Fixed by dfd09159ec981f42c7ced5620e536475013fef8e
